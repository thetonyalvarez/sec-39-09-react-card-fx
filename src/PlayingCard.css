.PlayingCard-image {
  width: 100%;
}