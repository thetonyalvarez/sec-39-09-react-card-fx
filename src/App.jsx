import React, { useState, useEffect } from 'react'
import axios from 'axios'
import Button from '@mui/material/Button'
import Container from '@mui/material/Container'
import Box from '@mui/material/Box'
import logo from './assets/bicycle-logo.png'
import PlayingCard from './PlayingCard'
import './App.css'

// Cards API
const BASE_API = "https://deckofcardsapi.com/api/deck/";
const SHUFFLE_API = "new/shuffle/?deck_count=1"

// Returns deck id
const getNewDeck = async () => {
  const resp = await axios.get(BASE_API + SHUFFLE_API)
  return resp.data.deck_id
}

let deck_id = await getNewDeck()

function App() {

  const [count, setCount] = useState(52)
  const [cards, setCards] = useState([])
  const [isRemaining, setIsRemaining] = useState(`Cards left in deck: ${count}`)
  const [disable, setDisable] = useState(false);

  async function fetchCard() {
    const response = await axios.get(BASE_API + `${deck_id}/draw/?count=1`)
    let { error, cards, remaining } = response.data

    if (error) {
      setIsRemaining(error)
      setDisable(true)
    } else {
      setCards(prevCards => [...prevCards, <PlayingCard key={cards[0].code} card={cards[0].image}/>])
      setCount(remaining)
      setIsRemaining(`Cards left in deck: ${count}`)
    }
  }

  useEffect(function fetchCardWhenMounted() {
    console.log('i fire once');

    fetchCard();
  }, [])

  return (
    <div className="App">
      <header className="App-header">
        <img className="App-logo" src={logo} alt="Blackjack!" />
      </header>
      <main className="App-main">
        <Container>
          <Box
            sx={{ my: 4 }}
          >
            <Button data-testid="deal-btn" color="secondary" variant="contained" onClick={fetchCard} disabled={disable}>
              Deal New Card
            </Button>
          </Box>
          <Box
            data-testid="isRemaining" 
            sx={{ my: 4 }}
          >
            {isRemaining}
          </Box>
          <Box
            data-testid="cards" 
            sx={{ display: 'inline' }}
          >
            {cards}
          </Box>
        </Container>
      </main>
    </div>
  )
}

export default App
