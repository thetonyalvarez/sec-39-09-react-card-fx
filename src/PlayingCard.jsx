import React from 'react';
import Box from '@mui/material/Box';
import './PlayingCard.css';

const PlayingCard = ({ card }) => {

  return (
    <Box data-testid="PlayingCard" sx={{ display: 'inline-block', width: 200 }}>
      <img className="PlayingCard-image" src={card} />
    </Box>
  )
}

export default PlayingCard 