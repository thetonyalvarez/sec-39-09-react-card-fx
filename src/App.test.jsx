import React from "react";
import { render, fireEvent, within } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import App from './App'
import PlayingCard from './PlayingCard'

// smoke test
it("renders without crashing", function() {
  render(<App />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("should render Deal New Card button", function() {
  const { getByTestId } = render(<App />)
  
  const dealBtn = getByTestId('deal-btn')
  const text = within(dealBtn).getByText('Deal New Card')

  expect(text).toBeVisible()
});

it("should render text declaring how many cards are left", function() {
  const { getByTestId } = render(<App />)
  
  const isRemaining = getByTestId('isRemaining')
  expect(isRemaining).toContainHTML('Cards left in deck')
});