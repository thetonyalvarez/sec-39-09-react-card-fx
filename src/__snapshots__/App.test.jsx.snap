// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="Blackjack!"
        class="App-logo"
        src="/src/assets/bicycle-logo.png"
      />
    </header>
    <main
      class="App-main"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-zitybv"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            data-testid="deal-btn"
            tabindex="0"
            type="button"
          >
            Deal New Card
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-zitybv"
          data-testid="isRemaining"
        >
          Cards left in deck: 52
        </div>
        <div
          class="MuiBox-root css-6n7j50"
          data-testid="cards"
        />
      </div>
    </main>
  </div>
</DocumentFragment>
`;
